#VRML_SIM R2021a utf8
WorldInfo {
  coordinateSystem "NUE"
}
Viewpoint {
  orientation 0.05111036404077116 0.997934503156858 -0.038916039067392084 1.3035232829320547
  position 0.2548631448572629 0.03580607077795888 -0.13866225551583702
}
TexturedBackground {
}
TexturedBackgroundLight {
}
RectangleArena {
  floorSize 5 5
  floorAppearance PBRAppearance {
    baseColor 0 0 0
    roughness 1
    roughnessMap ImageTexture {
      url [
        "textures/parquetry/chequered_parquetry_occlusion.jpg"
      ]
    }
    metalness 0
    normalMap ImageTexture {
      url [
        "textures/parquetry/chequered_parquetry_normal.jpg"
      ]
    }
    occlusionMap ImageTexture {
      url [
        "textures/parquetry/chequered_parquetry_occlusion.jpg"
      ]
    }
  }
}
DEF path Solid {
  translation 0.527 0 0.3
  children [
    DEF road Shape {
      appearance PBRAppearance {
        roughness 1
        metalness 0
      }
      geometry Box {
        size 0.9 0.0001 0.03
      }
    }
  ]
  name "path(4)"
  boundingObject USE road
}
DEF path1 Solid {
  translation -0.017 0 -0.129999
  rotation 0 1 0 -1.5707953071795862
  children [
    DEF road Shape {
      appearance PBRAppearance {
        roughness 1
        metalness 0
      }
      geometry Box {
        size 0.9 0.0001 0.03
      }
    }
  ]
  name "solid(3)"
}
Robot {
  translation -0.525292 0.03056 -0.0550869
  rotation -0.006826409078883563 0.0012920598256568677 -0.9999758650689997 -3.141575307179586
  children [
    DEF caster_left BallJoint {
      jointParameters BallJointParameters {
        anchor 0.056 0.016 -0.15
      }
      endPoint DEF left_caster Solid {
        translation 0.056016641909522456 0.015997705881768208 -0.14996588630965454
        rotation -0.12300669683510111 -0.9869351056943303 -0.10405983702581162 1.4172811641875531
        children [
          DEF caster Shape {
            appearance PBRAppearance {
            }
            geometry Sphere {
              radius 0.014
              subdivision 4
            }
          }
        ]
        name "caster_left"
        boundingObject USE caster
        physics Physics {
        }
      }
    }
    DEF caster_right BallJoint {
      jointParameters BallJointParameters {
        anchor -0.056 0.016 -0.15
      }
      endPoint DEF right_caster Solid {
        translation -0.056 0.016 -0.15
        rotation 0.0032692965634476445 0.9999704763584355 -0.006953999673843877 2.262677676521408
        children [
          USE caster
        ]
        name "caster_right"
        boundingObject USE caster
        physics Physics {
        }
      }
    }
    DEF body Shape {
      appearance PBRAppearance {
      }
      geometry Box {
        size 0.1 0.003 0.1
      }
    }
    DEF ir_panel Solid {
      translation -5.10335e-14 0 -0.075
      rotation 0 0 1 1.5708
      scale 1.5 1.5 1.5
      children [
        DistanceSensor {
          translation 0 -0.0275 0
          name "ir0"
          type "infra-red"
        }
        DistanceSensor {
          translation 0 -0.025 0
          name "ir1"
          type "infra-red"
        }
        DistanceSensor {
          translation 0 -0.02 0
          name "ir2"
          type "infra-red"
        }
        DistanceSensor {
          translation 0 -0.0075 0
          name "ir3"
          type "infra-red"
        }
        DistanceSensor {
          translation 0 0.0075 0
          name "ir4"
          type "infra-red"
        }
        DistanceSensor {
          translation 0 0.02 0
          name "ir5"
          type "infra-red"
        }
        DistanceSensor {
          translation 0 0.025 0
          name "ir6"
          type "infra-red"
        }
        DistanceSensor {
          translation 0 0.0275 0
          name "ir7"
          type "infra-red"
        }
        DEF ir_panel Shape {
          appearance PBRAppearance {
          }
          geometry Box {
            size 0.003 0.06 0.01
          }
        }
      ]
      name "ir_panel"
      boundingObject USE ir_panel
      physics Physics {
      }
    }
    DEF ts_panel Solid {
      translation 0 0 -0.095
      rotation 0 0 1 1.5708
      scale 2 2 2
      children [
        DistanceSensor {
          translation 0 -0.03 0
          name "ts0"
          type "infra-red"
        }
        DistanceSensor {
          translation 0 0.03 0
          name "ts1"
          type "infra-red"
        }
        USE ir_panel
      ]
      name "ts_pane;"
    }
    DEF axel_rear Solid {
      rotation 0 0 1 1.5708
      children [
        DEF axel Shape {
          appearance PBRAppearance {
          }
          geometry Cylinder {
            height 0.145
            radius 0.002
          }
        }
      ]
      name "axel"
      boundingObject USE axel
      physics Physics {
      }
    }
    DEF axel_front Solid {
      translation 0 0 -0.15
      rotation 0 0 1 1.5708
      children [
        DEF axel Shape {
          appearance PBRAppearance {
          }
          geometry Cylinder {
            height 0.145
            radius 0.002
          }
        }
      ]
      name "axel(1)"
      boundingObject USE axel
      physics Physics {
      }
    }
    LED {
      translation 0 -0.02 0
      rotation 1 0 0 3.14159
      children [
        Shape {
          appearance PBRAppearance {
            metalness 0
            emissiveIntensity 20
          }
          geometry Sphere {
            radius 0.005
            subdivision 3
          }
        }
      ]
    }
    DEF leftwheel HingeJoint {
      jointParameters HingeJointParameters {
        position 167.19061165518875
        anchor 0.075 0 0
      }
      device [
        PositionSensor {
          name "pos_left"
        }
        RotationalMotor {
          name "left wheel motor"
        }
        Brake {
          name "brake_left"
        }
      ]
      endPoint Solid {
        translation 0.0750045 0 0
        rotation 0.693637884000445 -0.6931447137615622 -0.1960022746393206 3.53155848047172
        children [
          DEF wheel Shape {
            appearance PBRAppearance {
              baseColor 1 0.980392 0.403922
              roughness 1
              metalness 0
            }
            geometry Cylinder {
              height 0.005
              radius 0.03
            }
          }
        ]
        name "left_wheel"
        boundingObject USE wheel
        physics DEF physics_wh Physics {
        }
      }
    }
    DEF rightwheel HingeJoint {
      jointParameters HingeJointParameters {
        position 173.77985016253544
        anchor -0.075 0 0
      }
      device [
        PositionSensor {
          name "pos_right"
        }
        Brake {
          name "brake_right"
        }
        RotationalMotor {
          name "right wheel motor"
        }
      ]
      endPoint Solid {
        translation -0.07500442134343432 -4.417691654060463e-06 1.0235807421180205e-06
        rotation 0.6795960782202973 -0.6723065669083791 -0.29351839901352145 3.7120652874126625
        children [
          USE wheel
        ]
        name "right_wheel"
        boundingObject USE wheel
        physics USE physics_wh
      }
    }
  ]
  boundingObject USE body
  physics Physics {
  }
  controller "line_follow"
}
DEF path Solid {
  translation -0.093 0 -0.58
  children [
    DEF road Shape {
      appearance PBRAppearance {
        roughness 1
        metalness 0
      }
      geometry Box {
        size 0.9 0.0001 0.03
      }
    }
  ]
  name "path"
  boundingObject USE road
}
DEF path Solid {
  translation -0.093 0 -1.47
  children [
    DEF road Shape {
      appearance PBRAppearance {
        roughness 1
        metalness 0
      }
      geometry Box {
        size 0.9 0.0001 0.03
      }
    }
  ]
  name "path(1)"
  boundingObject USE road
}
DEF path Solid {
  translation 0.427 0 0.3
  children [
    DEF road Shape {
      appearance PBRAppearance {
        roughness 1
        metalness 0
      }
      geometry Box {
        size 0.9 0.0001 0.03
      }
    }
  ]
  name "path(3)"
  boundingObject USE road
}
DEF path Solid {
  translation 0.527 0 -1.47
  children [
    DEF road Shape {
      appearance PBRAppearance {
        roughness 1
        metalness 0
      }
      geometry Box {
        size 0.9 0.0001 0.03
      }
    }
  ]
  name "path(2)"
  boundingObject USE road
}
DEF path1 Solid {
  translation -0.527 0 -0.129999
  rotation 0 1 0 -1.5707953071795862
  children [
    USE road
  ]
}
DEF path1 Solid {
  translation 0.943 0 0.450001
  rotation 0 1 0 -1.5707953071795862
  children [
    USE road
  ]
  name "solid(4)"
}
DEF path1 Solid {
  translation 0.943 0 -1.61
  rotation 0 1 0 -1.5707953071795862
  children [
    USE road
  ]
  name "solid(2)"
}
DEF path1 Solid {
  translation -0.527 0 -1.03
  rotation 0 1 0 -1.5707953071795862
  children [
    USE road
  ]
  name "solid(1)"
}
DEF ramp Transform {
  translation 0.94 0 -0.58
  rotation 0 1 0 -1.5707953071795862
  children [
    DEF rtop1 Solid {
      translation 0 0.15 0
      children [
        DEF rtop1 Shape {
          appearance PBRAppearance {
            metalness 0
          }
          geometry Box {
            size 0.35 0.0001 0.03
          }
        }
      ]
      name "rtop1"
      boundingObject USE rtop1
    }
    DEF rtop2 Solid {
      translation 0 0.15 0.088
      children [
        DEF rtop2 Shape {
          appearance PBRAppearance {
            metalness 0
          }
          geometry Box {
            size 0.03 0.0001 0.175
          }
        }
      ]
      name "rtop2"
    }
    DEF rpath1 Solid {
      translation -0.382 0.0742 0
      rotation 0 0 1 0.34906500000000046
      children [
        DEF rpath1 Shape {
          appearance PBRAppearance {
            metalness 0
          }
          geometry Box {
            size 0.438456 0.001 0.03
          }
        }
      ]
      name "rpath1"
      boundingObject USE rpath1
    }
    DEF rpath2 Solid {
      translation 0.381 0.0745 0
      rotation 0 0 1 -0.349065
      children [
        DEF rpath1 Shape {
          appearance PBRAppearance {
            metalness 0
          }
          geometry Box {
            size 0.438456 0.001 0.03
          }
        }
      ]
      name "rpath2"
      boundingObject USE rpath1
    }
    DEF rpath3 Transform {
      translation 0 0.0745 0.381
      rotation 1 0 0 0.349065
      children [
        Solid {
          rotation 0 1 0 1.5708
          children [
            USE rpath1
          ]
          name "rpath3"
          boundingObject USE rpath1
        }
      ]
    }
    DEF rampo Solid {
      translation -0.587 0 -0.175
      scale 0.001 0.001 0.001
      children [
        DEF SHAPE Shape {
          appearance PBRAppearance {
            baseColor 0 0 0
            roughness 1
            metalness 0
            name "DefaultMaterial"
          }
          geometry IndexedFaceSet {
            coord Coordinate {
              point [
                412 150 350, 412 5.55112e-14 350, 412 0 762, 762 5.55112e-14 350, 762 150 350, 762 0 762, 762 150 350, 412 150 350, 412 0 762, 762 0 762, 412 150 0, 762 150 0, 0 5.55112e-14 0, 1174 5.55112e-14 0, 412 5.55112e-14 350, 0 5.55112e-14 350, 0 5.55112e-14 0, 1174 5.55112e-14 0, 0 5.55112e-14 350, 412 150 350, 0 5.55112e-14 0, 412 150 0, 1174 5.55112e-14 350, 762 5.55112e-14 350, 762 150 0, 1174 5.55112e-14 350, 1174 5.55112e-14 0, 412 5.55112e-14 350, 412 150 350, 0 5.55112e-14 350, 412 0 762, 762 0 762, 1174 5.55112e-14 350, 762 150 350, 762 5.55112e-14 350, 762 150 350, 412 150 350, 762 150 0, 412 150 0, 762 150 350
              ]
            }
            normal Normal {
              vector [
                -1 0 0, -1 0 0, -1 0 0, 1 0 0, 1 0 0, 1 0 0, 0 0.93966 0.342109, 0 0.93966 0.342109, 0 0.93966 0.342109, 0 0.93966 0.342109, 0 0 -1, 0 0 -1, 0 0 -1, 0 0 -1, 0 -1 0, 0 -1 0, 0 -1 0, 0 -1 0, -0.342109 0.93966 0, -0.342109 0.93966 0, -0.342109 0.93966 0, -0.342109 0.93966 0, 0 -1 0, 0 -1 0, 0.342109 0.93966 0, 0.342109 0.93966 0, 0.342109 0.93966 0, 0 0 1, 0 0 1, 0 0 1, 0 -1 -1.34736e-16, 0 -1 -1.34736e-16, 0 0 1, 0 0 1, 0 0 1, 0.342109 0.93966 0, 0 1 0, 0 1 0, 0 1 0, 0 1 0
              ]
            }
            coordIndex [
              0, 1, 2, -1, 3, 4, 5, -1, 6, 7, 8, -1, 9, 6, 8, -1, 10, 11, 12, -1, 11, 13, 12, -1, 14, 15, 16, -1, 17, 14, 16, -1, 18, 19, 20, -1, 19, 21, 20, -1, 22, 23, 17, -1, 23, 14, 17, -1, 24, 25, 26, -1, 27, 28, 29, -1, 23, 30, 14, -1, 31, 30, 23, -1, 32, 33, 34, -1, 24, 35, 25, -1, 36, 37, 38, -1, 36, 39, 37, -1
            ]
          }
        }
      ]
      name ""
      boundingObject USE SHAPE
    }
  ]
}
